pkg/storage/watch/storage.go

```diff
@@ -0,0 +1,176 @@
+package watch
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	log "github.com/sirupsen/logrus"
+	api "github.com/weaveworks/ignite/pkg/apis/ignite"
+	"github.com/weaveworks/ignite/pkg/apis/ignite/scheme"
+	meta "github.com/weaveworks/ignite/pkg/apis/meta/v1alpha1"
+	"github.com/weaveworks/ignite/pkg/storage"
+	"github.com/weaveworks/ignite/pkg/storage/manifest/raw"
+	"github.com/weaveworks/ignite/pkg/storage/watch/update"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/yaml"
+)
+
+// Storage is an interface for persisting and retrieving API objects to/from a backend
+// One Storage instance handles all different Kinds of Objects
+type WatchStorage interface {
+	// WatchStorage extends the Storage interface
+	storage.Storage
+	// GetTrigger returns a hook that can be used to detect a watch event
```

Comment:
?

Reply:
Yes, the initial plan of having SyncStorage talk directly to the RawStorages didn't pan out very well, so I took the liberty of implementing WatchStorage. It sits as the Storage implementation between SyncStorage and MappedRawStorage and manages the watcher (`inotify`), the MappedStorage mappings and triggering SyncStorage to update via itself.